crates/bevy_render/src/texture/mod.rs

```diff
@@ -170,8 +174,176 @@ pub trait BevyDefault {
+    /// be available.
+    ///
+    /// Currently the only [`ViewTargetFormat`] that requires any features is [`Rb11b10Float`](ViewTargetFormat::Rb11b10Float).
+    pub const fn required_features(&self) -> wgpu::Features {
+        match self {
+            ViewTargetFormat::R8Unorm
+            | ViewTargetFormat::Rg8Unorm
+            | ViewTargetFormat::Bgra8Unorm
+            | ViewTargetFormat::Rgba8Unorm
+            | ViewTargetFormat::Rgba8UnormSrgb
+            | ViewTargetFormat::Bgra8UnormSrgb
+            | ViewTargetFormat::Rgba16Float
+            | ViewTargetFormat::R16Float
+            | ViewTargetFormat::Rg16Float
+            | ViewTargetFormat::Rgb10a2Unorm => wgpu::Features::empty(),
+            ViewTargetFormat::Rb11b10Float => wgpu::Features::RG11B10UFLOAT_RENDERABLE,
+        }
+    }
+
+    /// Unclamped [`ViewTargetFormat`]s are ones whose float values in shader won't be clamped between 0.0 and 1.0.
+    ///
+    /// Bloom and other such effects only work with these formats.
+    pub const fn is_unclamped(&self) -> bool {
+        // BEFORE_MERGE: Heavily bikeshed the name
+
+        match self {
+            ViewTargetFormat::R16Float
+            | ViewTargetFormat::Rg16Float
+            | ViewTargetFormat::Rgba16Float
+            | ViewTargetFormat::Rb11b10Float => true,
+            ViewTargetFormat::R8Unorm
+            | ViewTargetFormat::Rg8Unorm
+            | ViewTargetFormat::Rgba8Unorm
+            | ViewTargetFormat::Bgra8Unorm
+            | ViewTargetFormat::Rgba8UnormSrgb
+            | ViewTargetFormat::Bgra8UnormSrgb
+            | ViewTargetFormat::Rgb10a2Unorm => false,
+        }
+    }
+}
+
+impl From<ViewTargetFormat> for TextureFormat {
+    fn from(value: ViewTargetFormat) -> Self {
+        match value {
+            ViewTargetFormat::Rgba16Float => TextureFormat::Rgba16Float,
+            ViewTargetFormat::Rb11b10Float => TextureFormat::Rg11b10Float,
+            ViewTargetFormat::Rgba8UnormSrgb => TextureFormat::Rgba8UnormSrgb,
+            ViewTargetFormat::Rgba8Unorm => TextureFormat::Rgba8Unorm,
+            ViewTargetFormat::R8Unorm => TextureFormat::R8Unorm,
+            ViewTargetFormat::Rg8Unorm => TextureFormat::Rg8Unorm,
+            ViewTargetFormat::Bgra8Unorm => TextureFormat::Bgra8Unorm,
+            ViewTargetFormat::Bgra8UnormSrgb => TextureFormat::Bgra8UnormSrgb,
+            ViewTargetFormat::R16Float => TextureFormat::R16Float,
+            ViewTargetFormat::Rg16Float => TextureFormat::Rg16Float,
+            ViewTargetFormat::Rgb10a2Unorm => TextureFormat::Rgb10a2Unorm,
+        }
+    }
+}
+
+/// The error type returned when converting a [`TextureFormat`] into [`ViewTargetFormat`] fails.
+#[derive(Debug, Clone, Copy)]
+pub struct TryFromTextureFormatError {}
+
+impl Display for TryFromTextureFormatError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str("could not convert texture format into view target format")
```

Comment:
This error message could be more useful. Why aren't all texture formats suitable?